lib/JwsLinkedDataSignature.js

```diff
@@ -0,0 +1,188 @@
+    data.set(encodedHeader, 0);
+    data.set(verifyData, encodedHeader.length);
+
+    const signature = await this.signer.sign({data});
+
+    // create detached content signature
+    const encodedSignature = encode(signature);
+    proof.jws = encodedHeader + '..' + encodedSignature;
+    return proof;
+  }
+
+  /**
+   * @param verifyData {Uint8Array}.
+   * @param verificationMethod {object}.
+   * @param document {object} the document the proof applies to.
+   * @param proof {object} the proof to be verified.
+   * @param purpose {ProofPurpose}
+   * @param documentLoader {function}
+   * @param expansionMap {function}
+   *
+   * @returns {Promise<{boolean}>} Resolves with the verification result.
+   */
+  async verifySignature({verifyData, verificationMethod, proof}) {
+    if(!(proof.jws && typeof proof.jws === 'string' &&
+      proof.jws.includes('.'))) {
+      throw new TypeError('The proof does not include a valid "jws" property.');
+    }
+    // add payload into detached content signature
+    const [encodedHeader, /*payload*/, encodedSignature] = proof.jws.split('.');
+
+    let header;
+    try {
+      const utf8decoder = new TextDecoder();
+      header = JSON.parse(utf8decoder.decode(decode(encodedHeader)));
+    } catch(e) {
+      throw new Error('Could not parse JWS header; ' + e);
+    }
+    if(!(header && typeof header === 'object')) {
+      throw new Error('Invalid JWS header.');
+    }
+
+    // confirm header matches all expectations
+    if(!(header.alg === this.alg && header.b64 === false &&
+      Array.isArray(header.crit) && header.crit.length === 1 &&
+      header.crit[0] === 'b64') && Object.keys(header).length === 3) {
+      throw new Error(
+        `Invalid JWS header parameters for ${this.type}.`);
+    }
+
+    // do signature verification
+    const signature = decode(encodedSignature);
+
+    // concatenate the two uint8arrays
+    const data = new Uint8Array(encodedHeader.length + verifyData.length);
+    data.set(encodedHeader, 0);
+    data.set(verifyData, encodedHeader.length);
+
+    let {verifier} = this;
+    if(!verifier) {
+      const key = await this.LDKeyClass.from(verificationMethod);
+      verifier = key.verifier();
+    }
+    return verifier.verify({data, signature});
+  }
+
+  async assertVerificationMethod({verificationMethod}) {
+    if(!jsonld.hasValue(verificationMethod, 'type', this.requiredKeyType)) {
+      throw new Error(
+        `Invalid key type. Key type must be "${this.requiredKeyType}".`);
+    }
+  }
+
+  async getVerificationMethod({proof, documentLoader}) {
+    if(this.key) {
+      return this.key.export({publicKey: true});
+    }
+
+    const verificationMethod = await super.getVerificationMethod(
+      {proof, documentLoader});
+    await this.assertVerificationMethod({verificationMethod});
+    return verificationMethod;
+  }
+
+  async matchProof({proof, document, purpose, documentLoader, expansionMap}) {
+    if(!await super.matchProof(
```

Comment:
Note: Extending suites will need to check for their contexts here and in `sign`. We probably want a note about that somewhere.

Reply:
I added a note - does that work?